docs/tutorials/settings_tutorial.ipynb

```diff
@@ -0,0 +1,437 @@
+    "\n",
+    "Notice that we redefined `settings.agent.paper_directory` and `settings.agent.index` settings. `Paperqa` actually uses the setting from `settings.agent`. However, for convenience, we implemented an alias in `settings.paper_directory` and `settings.index_directory`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# The output\n",
+    "\n",
+    "`Paperqa` returns a `PQASession` object, which contains not only the answer but also all the information gatheres to answer the questions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response.__dict__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Let's examine the PQASession object returned by paperqa:\\n\")\n",
+    "\n",
+    "print(f\"Status: {response.status.value}\")\n",
+    "\n",
+    "print(\"1. Question asked:\")\n",
+    "print(f\"{response.session.question}\\n\")\n",
+    "\n",
+    "print(\"2. Answer provided:\")\n",
+    "print(f\"{response.session.answer}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In addition to the answer, the `PQASession` object contains all the references and contexts used to generate the answer.\n",
+    "\n",
+    "Because `paperqa` splits the documents into chunks, each chunk is a valid reference. You can see that it also references the page where the context was found."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"3. References cited:\")\n",
+    "print(f\"{response.session.references}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, `PQASession.session.contexts` contains the contexts used to generate the answer. Each context has a score, which is the similarity between the question and the context. \n",
+    "`Paperqa` uses this score to choose what contexts is more relevant to answer the question."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"4. Contexts used to generate the answer:\")\n",
+    "print(\"These are the relevant text passages that were retrieved and used to formulate the answer:\")\n",
+    "for i, ctx in enumerate(response.session.contexts, 1):\n",
+    "    print(f\"\\nContext {i}:\")\n",
+    "    print(f\"Source: {ctx.text.name}\")\n",
+    "    print(f\"Content: {ctx.context}\")\n",
+    "    print(f\"Score: {ctx.score}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
```

Comment:
We have a blank cell here, can we add `nb-clean` to `pre-commit` to auto-remove these:

```yaml
  - repo: https://github.com/srstevenson/nb-clean
    rev: 4.0.1
    hooks:
      - id: nb-clean
        args: [--preserve-cell-outputs, --remove-empty-cells]
```